libbeat/outputs/elasticsearch/client.go

```diff
@@ -45,24 +45,26 @@ type Client struct {
 	index    outputs.IndexSelector
 	pipeline *outil.Selector
 
-	observer outputs.Observer
+	observer           outputs.Observer
+	NonIndexableAction string
 
 	log *logp.Logger
 }
 
 // ClientSettings contains the settings for a client.
 type ClientSettings struct {
 	eslegclient.ConnectionSettings
-	Index    outputs.IndexSelector
-	Pipeline *outil.Selector
-	Observer outputs.Observer
+	Index              outputs.IndexSelector
+	Pipeline           *outil.Selector
+	Observer           outputs.Observer
+	NonIndexableAction string
 }
 
 type bulkResultStats struct {
 	acked        int // number of events ACKed by Elasticsearch
 	duplicates   int // number of events failed with `create` due to ID already being indexed
 	fails        int // number of failed events (can be retried)
-	nonIndexable int // number of failed events (not indexable -> must be dropped)
```

Comment:
Mayba add an extra counter for dropped events?

Reply:
This is a little wonky, nonIndexable turns into the dropped. (keeping the old logic) if a mapping error occurred, it's first a normal failure and only after the death letter index also fails is it deemed NonIndexable. We can keep better stats, but it would then get lost as downstream(libbeat/outputs/observer.go) there isn't a logical candidate for the extra stat. At least keeping the old logic allows the stats to be comparable between versions before / after merge of this PR. If someone needs the stats then I guess it's a new PR.